lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,7 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        raise Puppet::Error.new( "Puppet is not able to create or delete Active Directory users with the user resource", e ) if /8007089A in Active Directory/ =~ e.message
```

Comment:
The error message is likely localized and may not match the regex. Can you GetLastError instead? Or just match the error code?